src/components/Model3DViewer/Model3DViewer.tsx

```diff
@@ -235,21 +284,238 @@ export class Model3DViewer extends React.Component<Model3DViewerProps> {
+      if (sliceProps.x) {
+        const plane = new THREE.Plane(
+          new THREE.Vector3(sliceProps.x.direction ? 1 : -1, 0, 0),
+          sliceProps.x.coord
+        );
+        planes[0] = plane;
+      }
+      if (sliceProps.y) {
+        const plane = new THREE.Plane(
+          new THREE.Vector3(0, sliceProps.y.direction ? 1 : -1, 0),
+          sliceProps.y.coord
+        );
+        planes[1] = plane;
+      }
+      if (sliceProps.z) {
+        const plane = new THREE.Plane(
+          new THREE.Vector3(0, 0, sliceProps.z.direction ? 1 : -1),
+          sliceProps.z.coord
+        );
+        planes[2] = plane;
+      }
+      this.setState({ planes });
+    }
+  };
+
+  takeScreenShot = async () => {
+    if (this.viewer) {
+      const url = await this.viewer.getScreenshot();
+      if (this.props.onScreenshot) {
+        this.props.onScreenshot(url);
+      }
+    }
+  };
+
+  onchange = (val: number, axis: string) => {
+    const planes = this.state.planes;
+    const flipped = this.state.flipped;
+    switch (axis) {
+      case 'x': {
+        planes[0].set(planes[0].normal, flipped[0] ? val : -val);
+        break;
+      }
+      case 'y': {
+        planes[1].set(planes[1].normal, flipped[1] ? val : -val);
+        break;
+      }
+      case 'z': {
+        planes[2].set(planes[2].normal, flipped[2] ? val : -val);
+        break;
+      }
+      default: {
+        break;
+      }
+    }
+    this.setState({ planes });
+  };
+
+  flipSlider = (axis: string) => {
+    const planes = this.state.planes;
+    const flipped = this.state.flipped;
+    switch (axis) {
+      case 'x': {
+        planes[0].negate();
+        flipped[0] = !flipped[0];
+        break;
+      }
+      case 'y': {
+        planes[1].negate();
+        flipped[1] = !flipped[1];
+        break;
+      }
+      case 'z': {
+        planes[2].negate();
+        flipped[2] = !flipped[2];
+        break;
+      }
+      default: {
+        break;
+      }
+    }
+    this.setState({ planes });
+    this.setState({ flipped });
+  };
+
+  renderSliders = () => {
+    if (!this.props.slider) {
+      return <></>;
+    }
+    let xSlider = <></>;
+    let ySlider = <></>;
+    let zSlider = <></>;
+    if (this.props.slider.x) {
+      const x = this.props.slider.x;
+      xSlider = (
+        <div style={containerStyles}>
+          <span style={{ marginTop: '0.5vh' }}>
+            <h4>x</h4>
+          </span>
+          <Slider
+            step={(x.max - x.min) / 100}
+            min={x.min}
+            max={x.max}
+            defaultValue={x.max}
+            style={{ width: '18vw' }}
+            onChange={(val: SliderValue) => this.onchange(val as number, 'x')}
+          />
+          <Button
+            type="primary"
+            icon="redo"
+            size="small"
+            style={{ marginLeft: '2%' }}
+            onClick={() => this.flipSlider('x')}
+          />
+        </div>
+      );
+    }
+    if (this.props.slider.y) {
```

Comment:
Duplicate code as for x-axis. Refactor this

Reply:
Fixed